sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -231,6 +231,13 @@ class QueryExecution(val sparkSession: SparkSession, val logical: LogicalPlan) {
     """.stripMargin.trim
   }
 
+  /**
+   * Redact the sensitive information in the given string.
+   */
+  private def withRedaction(message: => String): String = {
+    Utils.redact(SparkSession.getActiveSession.map(_.sparkContext.conf).orNull, message)
```

Comment:
Why not make `Utils.redact` accept a regex parameter so that we can use session conf?

Reply:
A different query may need to use a different `regex`, a session conf will be better and the user can change it during runtime.